apps/percona/functional/snapshot_rebuild/test.yml

```diff
@@ -0,0 +1,189 @@
+            cstor_replicas_pod : []
+
+        - name: Obtaining the pool deployments from cvr
+          shell: >
+            kubectl get cvr -n {{ operator_ns }}
+            -l openebs.io/persistent-volume={{ pv.stdout }} --no-headers
+            -o=jsonpath='{range .items[*]}{.metadata.labels.cstorpool\.openebs\.io\/name}{"\n"}{end}'
+          args:
+            executable: /bin/bash
+          register: pool_deployment
+
+        - name: Obtaining the replicasets corresponding to pool deployements.
+          shell: >
+            kubectl get rs --selector=app=cstor-pool -n {{ operator_ns }} --no-headers
+            -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="{{item}}")].metadata.name}'
+          register: rs_list
+          with_items:
+            - "{{ pool_deployment.stdout_lines }}"
+
+        - name: Obtaining the pool pods
+          shell: >
+            kubectl get pod --selector=app=cstor-pool -n {{ operator_ns }} --no-headers
+            -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="{{item.stdout}}")].metadata.name}'
+          register: pool_pods
+          with_items:
+            - "{{ rs_list.results }}"
+
+        - name: Build a list of replica pods.
+          set_fact:
+            cstor_replicas_pod : "{{ cstor_replicas_pod }} + [ '{{ item.stdout }}' ]"
+          with_items: "{{ pool_pods.results }}"
+
+        # #Select any cstor replica pod to inject packet loss
+        - set_fact:
+            replica_pod: "{{ cstor_replicas_pod | list | first }}"
+
+        - block:
+
+        ## Inducing the packet drop on the above pool pod.
+            - include_tasks: "/common/utils/inject_packet_loss_tc.yml"
+              vars:
+                status: "induce"
+                target_pod: "{{ replica_pod }}"
+                operator_namespace: "{{ operator_ns }}"
+
+            - name: Check if the targeted replica is disconnected.
+              shell: >
+                kubectl exec -it {{ cstor_target.stdout }} -n {{ operator_ns }} --container  cstor-istgt 
+                -- istgtcontrol -q replica |json_pp |grep "\"replicaId" |awk -F ':' '{print $2}' | tr -d ',' |wc -l
+              register: connected_replica
+              until: "connected_replica.stdout == \"2\""
+              delay: 30
+              retries: 15
+        
+            - name: Creating snapshot.
+              include_tasks: /funclib/kubectl/k8s_snapshot_clone/create_snapshot.yml 
+              vars:
+                app_ns: "{{ namespace }}"
+                snapshot_name: "{{ snap_name }}"
+                pvc_name: "{{ app_pvc }}"
+
+            - name: Checking if the snapshot is not created in the degraded replica.
+              shell: kubectl exec -ti {{ replica_pod }} -n {{ operator_ns }} --container cstor-pool -- zfs list -t snapshot
+              args:
+                executable: /bin/bash
+              register: list_snap
+              until: "snap_name not in list_snap.stdout"
```

Comment:
This will be met in the first try itself right (if the functionality works..) is the intention to see that there is not snap on the degraded replica even after 30*10 sec? If so, we may need something of the type: `wait_for` followed by above check/task.

Reply:
instead of 'wait_for', probably, we better need to check for status of 'volumeSnapshot'

Reply:
volumeSnapshot & volumeSnapshotData are k8s CRs which hold minimal data (snap created or not) @vishnuitta -- I guess we are yet to add cstor replica level info into its spec...

Reply:
I overlooked this code, and, thought that create_snapshot.yml doesn't check for successful creation of snapshot.. but, tasks in create_snapshot.yml takes care of successful creation of snapshot. So, we are good here.
And, yes, if functionality works, this check should get succeeded with first iteration itself.